### PR TITLE
Azure pipelines - linux & compatibility testing

### PR DESCRIPTION
**Pull request by mike-north**
- Browser tests in firefox & chrome, across windows and linux
- TS & Ember-CLI compatibility test scenarios in Azure

### CODE REVIEW

**Comment by chriskrycho** [comment]
@mike-north have you been able to figure out why our Windows Azure builds keep timing out?

**Comment by mike-north** [comment]
@chriskrycho I'll need to get the azure folks involved to figure out what's going on. Sometimes the tests pass (i.e., the current azure linux build for this PR), sometimes the ember tests can't connect to headless chrome within the 30-second window, and sometimes our "skeleton app" can't be deleted in our mocha tests.

File: ts/tests/helpers/skeleton-app.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  teardown() {
+    if (this.watched) {
+      this.watched.kill();
+    }
+    this.tmpDir.removeCallback();
+  }
+
+  _ember(args: string[]) {
+    let ember = require.resolve('ember-cli/bin/ember');
+    return execa('node', [ember].concat(args), { cwd: this.root, cleanup: true });
```
Comment (mike-north) [inline ts/tests/helpers/skeleton-app.ts]: Potentially important change, `, cleanup: true` option ensures that killing the process also kills any child processes that it has spawned.

Comment (dfreeman) [reply ts/tests/helpers/skeleton-app.ts]: I think this option already [defaults to `true`](https://github.com/sindresorhus/execa#cleanup)

**Comment by mike-north** [comment]
> Is the plan to move entirely to Azure and drop Travis/Appveyor once this lands?

I know that ember-cli (and possibly ember) are investigating the feasibility of doing this. I think it's still a good idea to run them side-by-side for a while until we figure out what's best for this project

File: ts/tests/helpers/skeleton-app.ts (hunk 1 above)
Comment (mike-north) [reply ts/tests/helpers/skeleton-app.ts]: good catch!

**Comment by mike-north** [comment]
Merging based on @dfreeman's approval

**Pull request merged**